src/main/java/org/jetbrains/bsp/bazel/server/bsp/resolvers/ActionGraphResolver.java

```diff
@@ -26,9 +34,18 @@ public ActionGraphParser getActionGraphParser(List<String> targets, List<String>
               .withMnemonic(targets, languageIds)
               .executeBazelCommand();
 
-      ActionGraphContainer actionGraphContainer =
-          AnalysisProtos.ActionGraphContainer.parseFrom(process.getStdoutStream());
-      return new ActionGraphParser(actionGraphContainer);
+      ActionGraphParser actionGraphParser;
+      if (bazelData.getVersion().compareTo(ACTION_GRAPH_V2_VERSION) < 0) {
+        AnalysisProtos.ActionGraphContainer actionGraphContainer =
+            AnalysisProtos.ActionGraphContainer.parseFrom(process.getStdoutStream());
+        actionGraphParser = new ActionGraphV1Parser(actionGraphContainer);
```

Comment:
return statement can be here

Reply:
I'm not sure I understand. You prefer for me to have two different returns for each if branch?

Reply:
yes, assigning to `actionGraphParser` is redundant